backend/pkg/schema/service.go

```diff
@@ -224,3 +228,42 @@ func (s *Service) GetConfig() (*ConfigResponse, error) {
 func (s *Service) GetSubjectConfig(subject string) (*ConfigResponse, error) {
 	return s.registryClient.GetSubjectConfig(subject)
 }
+
+func (s *Service) Parse(schema *SchemaResponse) (avro.Schema, error) {
```

Comment:
Actually, I had to rollback the whole thing. The parse of the main reference needs to be call in the end after solving all references, otherwise it wouldn't resolve properly the schema

Reply:
and you still use the codec to parse decode the schema, so we need to return it :)

Reply:
This algo first goes to all leaves and parse them first, then it goes recursively to the parent and parse until we solved all the dependencies!

Reply:
Just looked into it in more detail. As far as I understand this solution works because the avro library uses a cache under the hood. So once you have parsed all the references in the library's internal cache it can resolve these on it's own. I think we should comment this, so that this is more obvious to someone reading the code.

Thus, I think we should also watch for errors when `avro.Parse()` all the references and return them if we hit any of these. I'll do a couple changes and push them into your branch. Are you able to test this with an actual payload that is serialized with such an avro schema? I plan to work on proper integration tests which will also cover this within the next few weeks - this is a weak spot in Console.

Reply:
@igormq Can you grant permissions to push into your branch? I think GitHub allows you tick a box which would allow Console maintainers to push code into your PR branch.

Reply:
That's it! I will add a comment about it

Reply:
Please, feel free to add anything needed